design/specs/navigation.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Navigation Updates
+description: 
+published: true
+date: 2022-08-25 18:02:53
+tags: 
+editor: markdown
+dateCreated: 2022-08-25 18:02:58
+---
+
+## Context
+
+Mathesar users must be able to navigate between pages and know where they are at any point while using the system.
+
+### Design Goals
+
+- Users clearly understand which page they're on
+- Users easily navigate to other pages
+- Users may anticipate the result of navigation interactions
+
+## Navigation structure diagram
+
+The navigation design will be based on the following structure:
+
+| 1st Level         | 2nd Level       | 3rd Level     | 4th Level   |
+|-------------------|-----------------|---------------|-------------|
+| Database Page     | Schema Page     | Table Page    | Record Page |
+|                   | Data Explorer   | Query Page    |             |
+|                   | Data Import     |               |             |
+
+- 1st level: Database
+- 2nd level: Schema
+- 3rd level: Entities
+- 4th level: Record
+
+[Navigation by Page](https://balsamiq.cloud/sjfrgln/pbe3tnu/r7FA3?f=N4IgUiBcCMA0IDkpxAYWfAMhkAhHAsjgFo4DSUA2gLoC%2BQA%3D)
+
+## Phasing Out Tab Navigation
+
+Tabs will be phased out as part of the new navigation improvements.
+
+## Navigation Methods
+
+Navigation within Mathesar will take place in three different ways:
+
+- Through the site header
+- Through links in pages
+- Direct access through page URLs
+
+### Navigation via site header
+
+- The logo in the site header will lead visitors to the site's root, in this case the database homepage.
+- A breadcrumb-like navigation will be displayed at the top part of the site's header as the user navigates through the pages.
+- When users click on a breadcrumb element, a [page-selector](#page-selector) component is displayed with a list of schemas or entities based on the breadcrumb level.
+- Upon reaching the record level, the top navigation will show a 'Go to Record' action, which will open the [Record Selector](/engineering/specs/record-selector) component.
+- The current page will not be included in the breadcrumb path.
+- The current page label will be placed underneath the top header component in a sub-header component.
+- An icon will be used to identify each type of page (table, schema, record etc.)
+- When we eventually add support for several databases, we'll need CRUD UI for databases, and we may design some additional UI that stays within the header at that point.
+
+### Navigation via links within pages
+
+Throughout Mathesar, the following links exist within pages which allow for more targeted navigation to other pages.
```

Comment:
Wouldn't there also be additional links to the Data Explorer from the Table Page? (e.g. to show a summarized view once the user groups something on the table page, or to build a query with filters applied on the Table Page)

Reply:
I wrote this section and I didn't include that info because I don't yet understand how that functionality is supposed to work. I think we can safely omit those details from this spec and either: (a) include them within the Data Explorer spec (oh what that's at 100% now, so maybe not), or handle that work within smaller more-detailed tickets. I don't want to get too hung up on putting every detail into this "Navigation via links within pages", as I'm sure we're likely to discover more opportunities for inter-page navigation as the product evolves.

Reply:
I think it's fine to defer to later specs. Perhaps add a line in this spec that this isn't meant to be comprehensive and other opportunities for inter-page navigation are expected to occur.